### DISCUSSION TITLE
Name of current Route

### DISCUSSION THREAD

**Discussion by henry5059** [leafsphp/leaf#192] [Q&A] [answered]
Hello, I am new using this framework, which I want to know how to get the name of the current route? Because if I found an option to return all the routes, but I only want the current one and excuse the inconvenience.

**Comment by mychidarko** [comment, marked as answer]
Hi @henry5059. You can call `getRoute` on the Leaf instance to get the current route:

```php
$route = app()->getRoute();
```

This will return an array containing the route path and method. You can access the route path using `['path']`

```php
$route = app()->getRoute();
$routePath = $route['path'];
```

**Reply by henry5059** [reply]
Ok, I really want to get the path as follows: `
{
  "path": "/try",
  "method": "GET",
  "name": "try_name" =< here
}`, 
but it only returns me this: `
{
  "path": "/try",
  "method": "GET"
}`. 
Which my idea is to work with permissions and roles. Unless there is another easier way, and excuse the trouble again.

**Reply by henry5059** [reply]
Hi again, according to my comment, I only refer to the name, but not the path or the method. So how do I get the name?

**Reply by mychidarko** [reply]
Leaf currently doesn't have an implementation for that. I've opened a new issue, it will be added as a new feature https://github.com/leafsphp/router/issues/3
